### ISSUE TITLE
error 422 when trying to pick image to insert in post.

### ISSUE THREAD

**Issue by inlinkz** [GrafiteInc/CMS#85]
When I try to insert image in a post, the image dialog remains blank and the server gives me a 422 error. 

Meanwhile, all this time, the server is at 100% cpu and at 30 seconds I get this:

`Symfony\Component\Debug\Exception\FatalErrorException: Maximum execution time of 30 seconds exceeded in /home/forge/example.com/releases/20170710131736/vendor/intervention/image/src/Intervention/Image/Gd/Encoder.php:34`

I have around 100 images and it is now completely impossible to create a blog post and pick an image. 

Can you please point me in the right direction ?

**Comment by mlantz** [comment]
I'll look into ways of solving this.

**Comment by mlantz** [comment]
I've looked into this and the latest version does a considerable amount of caching of images to ensure that they load promptly without heavy impact, even with the pagination set to 100. I've refactored things so the that the caches are set when you upload images which should improve overall performance, it also keeps the cached image now forever until the image is deleted or altered. Will release this weekend.
